csharp/src/Apache.Arrow/Arrays/ArrayData.cs

```diff
@@ -53,6 +60,26 @@ public sealed class ArrayData : IDisposable
             Offset = offset;
             Buffers = buffers;
             Children = children;
+
+            if (NullCount == RecalculateNullCount)
+            {
+                Recalculate();
+            }
+        }
+
+        private int Recalculate()
```

Comment:
No one is capturing this return value.

I would suggest leaving this code alone in this PR (it is large enough already). We can fix this case in a separate, dedicated PR.

Reply:
I've fixed in a branch but will revert in the PR.

Reply:
That's been pushed.